Initial commit

jakobw opened the pull request:
Adds the `valid-package-file-require` rule. This will detect
* missing file extension, i.e. report `require( './foo' )` as invalid
* missing `./` on relative paths which are required for resource loader package file requires, i.e. report `require( 'foo.js' )` as invalid (should be `require( './foo.js' )`.

jakobw committed:
Initial commit

Ladsgroup commented on README.md:
```diff
@@ -1,2 +1,5 @@
+This eslint plugin contains rules specific to MediaWiki's [ResourceLoader](https://www.mediawiki.org/wiki/ResourceLoader).
+
+## Rules
+* `valid-package-file-require` - ensures `require`d files contain the file extension and are proper relative paths, i.e. `./foo.js` instead of `foo.js`.
```
I would add a link to package modules page in mediawiki.org https://www.mediawiki.org/wiki/ResourceLoader/Package_modules
Finding it is hard and it contains lots of valuable information

jakobw replied:
Good point. Done!

jakobw committed:
Improve  rule definition

Ladsgroup reviewed (approved):
LGTM

jakobw committed:
Prefix the filne name with `./` before checking existence

This mitigates a difference between node versions where in one version
`require( 'foo.js' )` would be valid to require the file foo.js in the
same directory, whereas in another version it would error.
